src/liballoc/fmt.rs

```diff
@@ -97,7 +97,187 @@
 //! actual object being formatted, and the number of characters must have the
 //! type [`usize`].
```

Comment:
The section ending here ("Argument types") still sticks out to me. I basically cannot parse it, and I have no idea what it is trying to tell me.

The first sentence, "Each argument's type is dictated by the format string" is not even correct, is it? `{}` does not determine the type of the argument at all. Also this uses the `.*` syntax, which is *really subtle* in how it consumes two format arguments. That is explained way later though, in the subsection dedicated entirely to specifying the precision.

My personal feeling is that this subsection here does not add anything useful and is confusing. Hence I propose removing it.

Reply:
I agree. It could be left out here completely, and the bit about precision having to be `usize` is mentioned below already.